backend/api/v1/setting_service.go

```diff
@@ -1111,9 +1111,18 @@ func validateMaskingAlgorithm(algorithm *v1pb.MaskingAlgorithmSetting_Algorithm)
 		if algorithm.Mask == nil {
 			return nil
 		}
-		switch algorithm.Mask.(type) {
+		switch m := algorithm.Mask.(type) {
 		case *v1pb.MaskingAlgorithmSetting_Algorithm_FullMask_:
 		case *v1pb.MaskingAlgorithmSetting_Algorithm_RangeMask_:
+			for i, slice := range m.RangeMask.Slices {
```

Comment:
nit: might also be worth checking Start < End.

Reply:
Update